React/Views/RCTScrollView.m

```diff
@@ -289,8 +289,9 @@ - (void)dockClosestSectionHeader
   [_stickyHeaderIndices enumerateIndexesWithOptions:0 usingBlock:
    ^(NSUInteger idx, __unused BOOL *stop) {
 
+    // If the subviews are out of sync with the sticky header indices don't
+    // do anything.
     if (idx >= subviewCount) {
-      RCTLogError(@"Sticky header index %zd was outside the range {0, %zd}", idx, subviewCount);
```

Comment:
Maybe I could validate the indices in bridgeDidFinishTransaction so we can keep the warning if the indices passed are actually bad but not warn when they are out of sync when called from scrollViewDidScroll. I wanted to move the warning to js initially but the children array in js was pretty different that the reactSubviews in native (reactSubview seems to be flattened somehow).